evalml/automl/automl_search.py

```diff
@@ -86,6 +86,7 @@ def __init__(self,
                  tuner_class=None,
                  verbose=True,
                  optimize_thresholds=False,
+                 max_batches=None,
                  _max_batches=None):
```

Comment:
@christopherbunn because we named `_max_batches` as a "private" input arg, let's delete it in this PR in favor of `max_batches`. This is why we kept it private, to test it out before renaming to make it public!